server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -4927,10 +4927,10 @@ public void finalizeStop(VirtualMachineProfile profile, Answer answer) {
         if (owner.getState() == Account.State.disabled) {
             throw new PermissionDeniedException("The owner of " + vm + " is disabled: " + vm.getAccountId());
         }
-        if (VirtualMachineManager.ResoureCountRunningVMsonly.value()) {
+        if (!VirtualMachineManager.ResoureCountRunningVMsonly.value()) {
             // check if account/domain is with in resource limits to start a new vm
             ServiceOfferingVO offering = _serviceOfferingDao.findById(vm.getId(), vm.getServiceOfferingId());
-            resourceLimitCheck(owner, vm.isDisplayVm(), new Long(offering.getCpu()), new Long(offering.getRamSize()));
+            resourceLimitCheck(owner, vm.isDisplayVm(), Long.valueOf(offering.getCpu()), Long.valueOf(offering.getRamSize()));
```

Comment:
@Pearl1594 shouldn't this check be done irrespective of the `ResoureCountRunningVMsonly` value.
Server is trying to start the VM it should simply check resource limit while the limits may vary based on the setting. Or maybe I'm wrong here.

Reply:
@shwstppr I was wondering the same and I think @weizhouapache may be able to provide some insight here, I didn't quite understand the usage of 'resource.count.running.vms.only' - I have currently aligned it to how the checks have been done for other operations.

Reply:
@Pearl1594 @shwstppr 
if resource.count.running.vms.only is false, the resources (cpu/memory) of the vm have already been added to resource count of domain/account, when we create the vm. so it is not needed to check the resource limitation when we start the vm.

Reply:
Oh Ok.. Thanks @weizhouapache